### PR TITLE
decimal:fix bug caused by do_mul

### PR DESCRIPTION
**Pull request by AndreMouche**
## What have you changed? (mandatory)

This PR fix bug in the function `do_mul` of decimal,
The master will panic in the following case:
```
0.00000000000000*0.000000000000000000000000000000000000000000000000000000000000000
```

## What are the type of the changes? (mandatory)

- Bug fix (non-breaking change which fixes an issue)

## How has this PR been tested? (mandatory)

unit test

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

no

## Does this PR affect tidb-ansible update? (mandatory)

no

## Refer to a related PR or issue link (optional)

https://github.com/pingcap/tikv/pull/3190

The related code in tidb: https://github.com/pingcap/tidb/blob/master/types/mydecimal.go#L1725

### CODE REVIEW

File: src/coprocessor/codec/mysql/decimal.rs [hunk 1]
```diff
@@ -714,18 +714,18 @@ fn do_div_mod(
 /// `do_mul` multiplies two decimals.
 fn do_mul(lhs: &Decimal, rhs: &Decimal) -> Res<Decimal> {
     let (l_int_word_cnt, mut l_frac_word_cnt) = (
-        word_cnt!(lhs.int_cnt) as usize,
-        word_cnt!(lhs.frac_cnt) as usize,
+        i32::from(word_cnt!(lhs.int_cnt)),
```
Comment (BusyJay) [inline src/coprocessor/codec/mysql/decimal.rs]: Why change it to i32?

**Comment by coocood** [comment]
LGTM

**Comment by siddontang** [comment]
ping @AndreMouche

**Pull request merged**